aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/helpers/AuthHelper.kt

```diff
@@ -66,5 +67,20 @@ internal open class AuthHelper {
         fun getActiveUsername(username: String, alternateUsername: String?, userIDForSRP: String?): String {
             return alternateUsername ?: (userIDForSRP ?: username)
         }
+
+        /**
+         * This method is used to capture the username from the magic link.
+         * @param magicLinkCode the code from the magic link.
+         * @return the username from the magic link.
+         * */
+        fun getUsernameFromMagicLink(magicLinkCode: String): String? {
```

Comment:
Suggestion: To add Base64 decoding flag as a parameter which will add flexibility and allow it to handle different encoding schemes. This will make this method extensible for magicLink with different encoding schemes like: DEFAULT, NO_PADDING, URL_SAFE.

eg:
```kotlin
fun getUsernameFromMagicLink(magicLinkCode: String, base64Flag: Int = android.util.Base64.NO_WRAP): String? {
...

val jsonString = android.util.Base64.decode(header, base64Flag)
 
...
```